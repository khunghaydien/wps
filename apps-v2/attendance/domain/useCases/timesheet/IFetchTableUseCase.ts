import { DailyRecordDisplayFieldLayoutTable } from '@attendance/domain/models/DailyRecordDisplayFieldLayout';

import * as Interface from '@attendance/domain/useCases/IUseCase';

export type IInputData = Interface.IInputData<{
  employeeId?: string | null;
  startDate: string;
  endDate: string;
}>;

export type IOutputData = Interface.IOutputData<{
  employeeId: string;
  startDate: string;
  endDate: string;
  layoutTable: DailyRecordDisplayFieldLayoutTable;
}>;

export type IUseCase = Interface.IUseCase<IInputData, IOutputData>;
