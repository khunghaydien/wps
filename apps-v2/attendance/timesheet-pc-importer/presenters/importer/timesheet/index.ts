import check from './check';
import fetchContractedWorkTimes from './fetchContractedWorkTimes';
import save from './save';
import { bind } from '@attendance/libraries/Collection';
import { AppStore } from '@attendance/timesheet-pc-importer/store/AppStore';

export default (store: AppStore) =>
  bind({ check, fetchContractedWorkTimes, save }, store);
