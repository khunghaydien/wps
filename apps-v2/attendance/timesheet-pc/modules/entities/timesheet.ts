// FIXME: 本来ならばドメインの型を使用するべきだが古い実装のために仕方なく使用している
import { createDailyContractDetail as createAttDailyContractDetailFromRemote } from '@attendance/repositories/models/DailyRecord';
import { Timesheet as TimesheetFromRemote } from '@attendance/repositories/models/Timesheet';

import AttRecord from '../../models/AttRecord';
import { createAttDailyAttentions } from '@attendance/domain/models/AttDailyAttention';
import { AttDailyRecordContractedDetail } from '@attendance/domain/models/AttDailyRecord';
import * as AttDailyRequest from '@attendance/domain/models/AttDailyRequest';
import * as AttDailyRequestType from '@attendance/domain/models/AttDailyRequestType';
import { detectPerformableActionForFix } from '@attendance/domain/models/AttFixSummaryRequest';
import { CommuteCount } from '@attendance/domain/models/CommuteCount';
import { DailyObjectivelyEventLog } from '@attendance/domain/models/DailyObjectivelyEventLog';
import { AttSummary } from '@attendance/domain/models/DeprecatedAttSummary';
import { createFromRemote as createFixDailyRequestFromRemote } from '@attendance/domain/models/FixDailyRequest';
import { OwnerInfo, Period } from '@attendance/domain/models/Timesheet';
import * as WorkingType from '@attendance/domain/models/WorkingType';

import ROOT from './actionType';

import * as RecordsUtil from '@attendance/libraries/utils/Records';

import * as attendanceHelper from '@attendance/ui/helpers/attentionDailyMessages';

const ACTION_TYPE_ROOT = `${ROOT}/TIMESHEET` as const;

export type State = {
  ownerInfos: OwnerInfo[] | null;
  workingTypes: WorkingType.WorkingType[] | null;
  summaryPeriodList: Period[] | null;
  attDailyRequestTypeMap: AttDailyRequestType.DailyRequestNameMap | null;
  attDailyRequestMap: {
    [key: string]: AttDailyRequest.AttDailyRequest;
  } | null;
  attRecordList: Array<AttRecord> | null;
  dailyAttentionMessagesMap: { [date: string]: string[] } | null;
  dailyContractedDetailMap: {
    [date: string]: AttDailyRecordContractedDetail;
  } | null;
  attSummary: AttSummary | null;
  dailyObjectivelyEventLogs: DailyObjectivelyEventLog[] | null;
  workingType: {
    useAllowanceManagement: boolean;
    useManageCommuteCount: boolean;
    useObjectivelyEventLog: boolean;
    useFixDailyRequest: boolean;
    useLegalAgreement: boolean;
    useLegalAgreementMonthlyRequest: boolean;
    useLegalAgreementYearlyRequest: boolean;
  };
  dailyRestCountLimit: number;
};

export const initialState: State = {
  ownerInfos: null,
  workingTypes: null,
  summaryPeriodList: null,
  attDailyRequestTypeMap: null,
  attDailyRequestMap: null,
  attRecordList: null,
  dailyAttentionMessagesMap: null,
  dailyContractedDetailMap: null,
  attSummary: null,
  dailyObjectivelyEventLogs: null,
  workingType: {
    useLegalAgreement: false,
    useLegalAgreementMonthlyRequest: false,
    useLegalAgreementYearlyRequest: false,
    useAllowanceManagement: false,
    useManageCommuteCount: false,
    useObjectivelyEventLog: false,
    useFixDailyRequest: false,
  },
  dailyRestCountLimit: 0,
};

const WORKING_TYPE_KEYS = Object.keys(initialState.workingType) as Array<
  keyof State['workingType']
>;

export const ACTIONS = {
  SET_TIMESHEET_ITEMS: `${ACTION_TYPE_ROOT}/SET_TIMESHEET_ITEMS`,
  SET_DAILY_OBJECTIVELY_EVENT_LOGS: `${ACTION_TYPE_ROOT}/SET_DAILY_OBJECTIVELY_EVENT_LOGS`,
  UPDATE_COMMUTE_COUNT: `${ACTION_TYPE_ROOT}/UPDATE_COMMUTE_COUNT`,
  UPDATE_DAILY_OBJECTIVELY_EVENT_LOG: `${ACTION_TYPE_ROOT}/UPDATE_DAILY_OBJECTIVELY_EVENT_LOG`,
  CLEAR_REQUEST_TYPE_CODES: `${ACTION_TYPE_ROOT}/CLEAR_REQUEST_TYPE_CODES`,
  SET_REQUEST_TYPE_CODES: `${ACTION_TYPE_ROOT}/SET_REQUEST_TYPE_CODES`,
} as const;

type SetTimesheetAction = {
  type: typeof ACTIONS.SET_TIMESHEET_ITEMS;
  payload: {
    entities: TimesheetFromRemote;
    targetEmployeeId: string | null;
    dailyObjectivelyEventLogs?: DailyObjectivelyEventLog[] | null;
  };
};

type UpdateCommuteCountAction = {
  type: typeof ACTIONS.UPDATE_COMMUTE_COUNT;
  payload: {
    targetDate: string;
    commuteCount: CommuteCount;
  };
};

type SetDailyObjectivelyEventLogsAction = {
  type: typeof ACTIONS.SET_DAILY_OBJECTIVELY_EVENT_LOGS;
  payload: DailyObjectivelyEventLog[] | null;
};

type UpdateDailyObjectivelyEventLogAction = {
  type: typeof ACTIONS.UPDATE_DAILY_OBJECTIVELY_EVENT_LOG;
  payload: {
    targetDate: string;
    dailyObjectivelyEventLog: DailyObjectivelyEventLog | null;
  };
};

type ClearRequestTypeCodes = {
  type: typeof ACTIONS.CLEAR_REQUEST_TYPE_CODES;
};

type SetRequestTypeCodes = {
  type: typeof ACTIONS.SET_REQUEST_TYPE_CODES;
  payload: { [id: string]: AttDailyRequestType.Code[] };
};

type Actions =
  | SetTimesheetAction
  | SetDailyObjectivelyEventLogsAction
  | UpdateCommuteCountAction
  | UpdateDailyObjectivelyEventLogAction
  | ClearRequestTypeCodes
  | SetRequestTypeCodes;

export const actions = {
  setTimesheetItems: (
    entities: TimesheetFromRemote,
    targetEmployeeId: string | null = null
  ): SetTimesheetAction => ({
    type: ACTIONS.SET_TIMESHEET_ITEMS,
    payload: {
      entities,
      targetEmployeeId,
    },
  }),
  setDailyObjectivelyEventLogs: (
    records: DailyObjectivelyEventLog[] | null
  ): SetDailyObjectivelyEventLogsAction => ({
    type: ACTIONS.SET_DAILY_OBJECTIVELY_EVENT_LOGS,
    payload: records,
  }),
  updateCommuteCount: (
    targetDate: string,
    commuteCount: CommuteCount
  ): UpdateCommuteCountAction => ({
    type: ACTIONS.UPDATE_COMMUTE_COUNT,
    payload: {
      targetDate,
      commuteCount,
    },
  }),
  updateDailyObjectivelyEventLog: (record: {
    targetDate: string;
    dailyObjectivelyEventLog: DailyObjectivelyEventLog | null;
  }): UpdateDailyObjectivelyEventLogAction => ({
    type: ACTIONS.UPDATE_DAILY_OBJECTIVELY_EVENT_LOG,
    payload: record,
  }),
  clearRequestTypeCodes: (): ClearRequestTypeCodes => ({
    type: ACTIONS.CLEAR_REQUEST_TYPE_CODES,
  }),
  setRequestTypeCodes: (requestTypeCodesMap: {
    [id: string]: AttDailyRequestType.Code[];
  }): SetRequestTypeCodes => ({
    type: ACTIONS.SET_REQUEST_TYPE_CODES,
    payload: requestTypeCodesMap,
  }),
};

/**
 * 日毎の注意喚起メッセージを作成する
 */
const createDailyAttentionMessages = (
  record: Parameters<typeof createAttDailyAttentions>[0]
): string[] => {
  return attendanceHelper.alert(createAttDailyAttentions(record));
};

const createDailyAttentionMessageMap = ({
  attRecordList,
  dailyObjectivelyEventLogs,
}: {
  attRecordList: State['attRecordList'];
  dailyObjectivelyEventLogs: State['dailyObjectivelyEventLogs'];
}) =>
  attRecordList.reduce((hash, attRecord) => {
    hash[attRecord.recordDate] = createDailyAttentionMessages({
      ...attRecord,
      objectivelyEventLog: dailyObjectivelyEventLogs?.find(
        (eventLog) => eventLog.recordId === attRecord.id
      ),
    });
    return hash;
  }, {});

export default function reducer(
  state: State = initialState,
  action: Actions
): State {
  switch (action.type) {
    case ACTIONS.SET_TIMESHEET_ITEMS:
      const { entities } = action.payload;
      const dailyObjectivelyEventLogs = state.dailyObjectivelyEventLogs;
      const ownerInfos: OwnerInfo[] = entities.employeeInfoList.map((data) => ({
        startDate: data.startDate,
        endDate: data.endDate,
        employee: {
          code: '',
          name: entities.employeeName,
        },
        department: {
          name: data.departmentName,
        },
        workingType: {
          name: data.workingTypeName,
        },
      }));
      const workingTypes = entities.workingTypeList.map(
        WorkingType.createFromRemote
      );
      const workingType = WORKING_TYPE_KEYS.reduce(
        (obj, key) => {
          obj[key] = entities.workingTypeList.some((wt) => wt[key]);
          return obj;
        },
        {
          useAllowanceManagement: false,
          useObjectivelyEventLog: false,
          useManageCommuteCount: false,
          useFixDailyRequest: false,
          useLegalAgreement: false,
          useLegalAgreementMonthlyRequest: false,
          useLegalAgreementYearlyRequest: false,
        } as State['workingType']
      );
      const lastDayWorkingType = workingTypes.at(-1);
      workingType.useLegalAgreementMonthlyRequest =
        lastDayWorkingType.useLegalAgreementMonthlyRequest;
      workingType.useLegalAgreementYearlyRequest =
        lastDayWorkingType.useLegalAgreementYearlyRequest;
      workingType.useLegalAgreement =
        workingType.useLegalAgreementMonthlyRequest ||
        workingType.useLegalAgreementYearlyRequest;

      const summaryPeriodList = entities.periods;
      const attDailyRequestTypeMap = entities.requestTypes;
      const attDailyRequestMap = Object.keys(entities.requests || {}).reduce(
        (hash, key) => {
          const request = entities.requests[key];
          hash[key] = AttDailyRequest.createFromRemote(
            attDailyRequestTypeMap,
            request,
            entities
          );
          return hash;
        },
        {}
      );
      const attSummary = {
        id: entities.id,
        requestId: entities.requestId,
        status: entities.status,
        approver01Name: entities.approver01Name,
        isLocked: entities.isLocked,
        isAllLeaveOfAbsence: entities.isAllAbsent,
        performableActionForFix: detectPerformableActionForFix(entities.status),
      };
      const attRecordList = entities.records.map((record) => {
        return AttRecord.createFromParam(
          {
            ...record,
            fixDailyRequest: createFixDailyRequestFromRemote(
              record,
              entities.records
            ),
            outInsufficientMinimumWorkHours:
              record.isFlexWithoutCoreRequireEarlyLeaveApply &&
              record.isFlexWithoutCore
                ? record.outInsufficientMinimumWorkHours
                : 0,
          },
          {
            isSummaryLocked: attSummary?.isLocked,
            useFixDailyRequest: RecordsUtil.getWithinRange(
              record.recordDate,
              workingTypes
            )?.useFixDailyRequest,
          }
        );
      });
      const dailyAttentionMessagesMap = createDailyAttentionMessageMap({
        attRecordList,
        dailyObjectivelyEventLogs,
      });
      const dailyContractedDetailMap = attRecordList.reduce(
        (hash, attRecord) => {
          hash[attRecord.recordDate] = attRecord.contractedDetail
            ? createAttDailyContractDetailFromRemote(attRecord.contractedDetail)
            : null;
          return hash;
        },
        {}
      );
      return {
        ...state,
        ownerInfos,
        workingTypes,
        workingType,
        summaryPeriodList,
        attDailyRequestTypeMap,
        attDailyRequestMap,
        attRecordList,
        dailyAttentionMessagesMap,
        dailyContractedDetailMap,
        attSummary,
        dailyObjectivelyEventLogs,
        dailyRestCountLimit: entities.dailyRestCountLimit,
      };

    case ACTIONS.SET_DAILY_OBJECTIVELY_EVENT_LOGS: {
      const dailyObjectivelyEventLogs = action.payload;
      const { attRecordList } = state;

      const dailyAttentionMessagesMap = createDailyAttentionMessageMap({
        attRecordList,
        dailyObjectivelyEventLogs,
      });
      return {
        ...state,
        dailyAttentionMessagesMap,
        dailyObjectivelyEventLogs,
      };
    }

    case ACTIONS.UPDATE_DAILY_OBJECTIVELY_EVENT_LOG: {
      const { targetDate, dailyObjectivelyEventLog } = action.payload;
      const prevRecords = state.dailyObjectivelyEventLogs.filter(
        ({ recordDate }) => recordDate !== targetDate
      );
      const dailyObjectivelyEventLogs = dailyObjectivelyEventLog
        ? prevRecords.concat([dailyObjectivelyEventLog])
        : prevRecords;
      const { attRecordList } = state;
      const dailyAttentionMessagesMap = createDailyAttentionMessageMap({
        attRecordList,
        dailyObjectivelyEventLogs,
      });
      return {
        ...state,
        dailyAttentionMessagesMap,
        dailyObjectivelyEventLogs,
      };
    }

    case ACTIONS.UPDATE_COMMUTE_COUNT: {
      const { targetDate, commuteCount } = action.payload;
      const { attRecordList: $attRecordList, attSummary, workingTypes } = state;
      const attRecordList = $attRecordList.map((record) => {
        if (record.recordDate === targetDate) {
          return new AttRecord({
            ...record,
            commuteCount,
            isSummaryLocked: attSummary?.isLocked,
            useFixDailyRequest: RecordsUtil.getWithinRange(
              record.recordDate,
              workingTypes
            )?.useFixDailyRequest,
          });
        } else {
          return record;
        }
      });

      return {
        ...state,
        attRecordList,
      };
    }

    case ACTIONS.CLEAR_REQUEST_TYPE_CODES: {
      const { attRecordList: $attRecordList, workingTypes, attSummary } = state;
      const attRecordList = $attRecordList.map(
        (record) =>
          new AttRecord({
            ...record,
            requestTypeCodes: null,
            isSummaryLocked: attSummary?.isLocked,
            useFixDailyRequest: RecordsUtil.getWithinRange(
              record.recordDate,
              workingTypes
            )?.useFixDailyRequest,
          })
      );

      return {
        ...state,
        attRecordList,
      };
    }

    case ACTIONS.SET_REQUEST_TYPE_CODES: {
      const { attRecordList: $attRecordList, workingTypes, attSummary } = state;
      const requestTypeCodesMap = action.payload;
      const attRecordList = $attRecordList.map(
        (record) =>
          new AttRecord({
            ...record,
            requestTypeCodes: requestTypeCodesMap[record.id] || [],
            isSummaryLocked: attSummary?.isLocked,
            useFixDailyRequest: RecordsUtil.getWithinRange(
              record.recordDate,
              workingTypes
            )?.useFixDailyRequest,
          })
      );

      return {
        ...state,
        attRecordList,
      };
    }

    default:
      return state;
  }
}
