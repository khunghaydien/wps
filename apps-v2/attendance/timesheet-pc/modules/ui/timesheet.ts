// State

type State = {
  isManHoursGraphOpened: boolean;
};

const initialState: State = {
  isManHoursGraphOpened: false,
};

// Actions

type ToggleManHoursGraph = {
  type: 'TIMESHEET-PC/UI/TIMESHEET/TOGGLE_MAN_HOURS_GRAPH';
};

type Action = ToggleManHoursGraph;

const ACTION_TYPES = {
  TOGGLE_MAN_HOURS_GRAPH: 'TIMESHEET-PC/UI/TIMESHEET/TOGGLE_MAN_HOURS_GRAPH',
} as const;

export const actions = {
  toggleManHoursGraph: (): ToggleManHoursGraph => ({
    type: ACTION_TYPES.TOGGLE_MAN_HOURS_GRAPH,
  }),
};

// Reducer

export default function reducer(
  state: State = initialState,
  action: Action
): State {
  switch (action.type) {
    case ACTION_TYPES.TOGGLE_MAN_HOURS_GRAPH:
      return {
        ...state,
        isManHoursGraphOpened: !state.isManHoursGraphOpened,
      };

    default:
      return state;
  }
}
