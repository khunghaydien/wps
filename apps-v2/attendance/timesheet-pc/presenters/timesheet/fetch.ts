import { Store } from 'redux';

import {
  catchApiError,
  catchBusinessError,
  loadingEnd,
  loadingStart,
} from '@apps/commons/actions/app';
import msg from '@apps/commons/languages';

import { State } from '@attendance/timesheet-pc/modules';
import { actions as selectedPeriodUiActions } from '@attendance/timesheet-pc/modules/client/selectedPeriodStartDate';
import { actions as entitiesActions } from '@attendance/timesheet-pc/modules/entities/timesheet';

import { IPresenter } from '@attendance/application/useCaseInteractors/timesheet/FetchUseCaseInteractor';

export default (store: Store): IPresenter => ({
  start: () => {
    store.dispatch(loadingStart());
  },
  complete: (result) => {
    const { timesheet, employeeId } = result;
    if (timesheet?.isMigratedSummary) {
      store.dispatch(
        catchBusinessError(
          msg().Com_Lbl_Error,
          msg().Att_Err_CanNotDisplayBeforeUsing,
          null,
          {
            isContinuable: false,
          },
          false
        )
      );
      return;
    }
    store.dispatch(entitiesActions.setTimesheetItems(timesheet, employeeId));
    store.dispatch(selectedPeriodUiActions.set(timesheet?.startDate));
  },
  error: (err) => {
    if (!(store.getState() as State).common.app.error) {
      store.dispatch(catchApiError(err as Parameters<typeof catchApiError>[0]));
    }
  },
  finally: () => {
    store.dispatch(loadingEnd());
  },
});
