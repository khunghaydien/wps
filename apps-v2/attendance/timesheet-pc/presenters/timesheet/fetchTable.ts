import { Store } from 'redux';

import { catchApiError } from '@apps/commons/actions/app';

import { actions } from '@attendance/timesheet-pc/modules/ui/dailyRecordDisplayFieldLayout';

import { IPresenter } from '@attendance/application/useCaseInteractors/timesheet/FetchTableUseCaseInteractor';

export default ({ dispatch }: Store): IPresenter => ({
  start: () => {
    dispatch(actions.reset());
    dispatch(actions.startLoading());
  },
  complete: (response) => {
    dispatch(actions.set(response.layoutTable));
  },
  error: (err) => {
    dispatch(actions.reset());
    dispatch(actions.catchError());
    dispatch(
      catchApiError(err as Parameters<typeof catchApiError>[0], {
        isContinuable: true,
      })
    );
  },
  finally: () => {
    dispatch(actions.endLoading());
  },
});
