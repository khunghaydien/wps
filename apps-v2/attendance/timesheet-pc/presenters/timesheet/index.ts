import { Store } from 'redux';

import fetch from './fetch';
import fetchTable from './fetchTable';

export default (
  store: Store
): {
  fetch: ReturnType<typeof fetch>;
  fetchTable: ReturnType<typeof fetchTable>;
} => ({
  fetch: fetch(store),
  fetchTable: fetchTable(store),
});
