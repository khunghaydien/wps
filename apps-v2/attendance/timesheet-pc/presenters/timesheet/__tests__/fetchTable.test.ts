import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import presenter from '../fetchTable';
import { IOutputData } from '@attendance/domain/useCases/timesheet/IFetchTableUseCase';

const mockStore = configureMockStore([thunk]);

beforeEach(() => {
  jest.clearAllMocks();
});

describe('reload', () => {
  describe('complete()', () => {
    it('should do.', () => {
      // Arrange
      const store = mockStore({});

      // Act
      presenter(store).complete({
        startDate: '2022-12-01',
        endDate: '2022-12-31',
        layoutTable: 'layoutTable',
      } as unknown as IOutputData);

      // Assert
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('error()', () => {
    it('should do.', () => {
      // Arrange
      const store = mockStore({});

      // Act
      presenter(store).error({
        message: 'Error Test',
      });

      // Assert
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('start()', () => {
    it('should do.', () => {
      // Arrange
      const store = mockStore({});

      // Act
      presenter(store).start();

      // Assert
      expect(store.getActions()).toMatchSnapshot();
    });
  });

  describe('finally()', () => {
    it('should do.', () => {
      // Arrange
      const store = mockStore({});

      // Act
      presenter(store).finally();

      // Assert
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
