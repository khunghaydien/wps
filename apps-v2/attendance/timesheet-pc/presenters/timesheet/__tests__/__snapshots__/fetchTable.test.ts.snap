// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reload complete() should do. 1`] = `
Array [
  Object {
    "payload": "layoutTable",
    "type": "TIMESHEET-PC/UI/DAILY_FILED_LAYOUT/SET",
  },
]
`;

exports[`reload error() should do. 1`] = `
Array [
  Object {
    "type": "TIMESHEET-PC/UI/DAILY_FILED_LAYOUT/RESET",
  },
  Object {
    "type": "TIMESHEET-PC/UI/DAILY_FILED_LAYOUT/CATCH_ERROR",
  },
  Object {
    "payload": Object {
      "errorCode": undefined,
      "isContinuable": true,
      "isFunctionCantUseError": false,
      "message": "APIエラー - Error Test",
      "problem": "Error Test",
      "stackTrace": "",
      "type": "APIエラー",
    },
    "type": "CATCH_API_ERROR",
  },
]
`;

exports[`reload finally() should do. 1`] = `
Array [
  Object {
    "type": "TIMESHEET-PC/UI/DAILY_FILED_LAYOUT/END_LOADING",
  },
]
`;

exports[`reload start() should do. 1`] = `
Array [
  Object {
    "type": "TIMESHEET-PC/UI/DAILY_FILED_LAYOUT/RESET",
  },
  Object {
    "type": "TIMESHEET-PC/UI/DAILY_FILED_LAYOUT/START_LOADING",
  },
]
`;
